src/main/java/org/commcare/formplayer/application/UtilController.java

```diff
@@ -102,7 +106,12 @@ public NotificationMessage deleteApplicationDbs(
     @UserLock
     public NotificationMessage clearUserData(
             @RequestBody AuthenticatedRequestBean requestBean,
-            HttpServletRequest request) {
+            HttpServletRequest request) throws InvalidStructureException {
+
+        // Test clearing case db
+        CaseSearchDB caseSearchDB = new CaseSearchDB(requestBean.getDomain(), requestBean.getUsername(),
+                requestBean.getRestoreAs());
+        caseSearchDB.deleteDatabaseFile();
```

Comment:
This introduces a new error handling path which will invert control out of the method since it's added to the method signature. That means that an issue with the database clearing will theoretically block the existing functionality. 

I'm fine with the return path of the InvalidStructureException causing a hard fault (that's a very unexpected result from the code), but I'd prefer for the new lines to be placed after the old ones, so a failure in the new functionality is an at-worst-neutral change for users

Reply:
moved new lines here https://github.com/dimagi/formplayer/pull/1539/commits/02fcbc457d72b791746fe55ab720a044feb4fba5